src/ipvs/ip_vs_service.c

```diff
@@ -218,7 +244,7 @@ __dp_vs_svc_match_get(int af, const struct rte_mbuf *mbuf)
         if (svc->af == af && svc->proto == iph->next_proto_id &&
             __svc_in_range(af, &saddr, ports[0], &m->srange) &&
             __svc_in_range(af, &daddr, ports[1], &m->drange) &&
-            (!idev || idev->id == mbuf->port) &&
+            (!idev || idev->id == iif) &&
```

Comment:
Tunnel device is just a case for the problem only. Other virtual devices, such as vlan device, have different device id from the  id of `mbuf->port`.  This is not a completed fix.

Reply:
添加了vlan虚拟端口的处理。

Reply:
Sorry, i was not aware `mbuf->id` is translated to the virtual device's port id in its receive function(i.e.  `p_tunnel_rcv`, `vlan_rcv`) just now. I think it can be matched anyway if `m->iifname` is set to tunnel/vlan device's name.  Use physical device name in `m->iifname` instead of virtual device name is not easy to understand, and may arise safety problem, even if it simplifies the config work when a lot many virtual devices existing. I do not think it needs to change here.